project/context.go

```diff
@@ -19,23 +19,25 @@ var projectRegexp = regexp.MustCompile("[^a-zA-Z0-9_.-]")
 // Context holds context meta information about a libcompose project, like
 // the project name, the compose file, etc.
 type Context struct {
-	ComposeFiles        []string
-	ComposeBytes        [][]byte
-	ProjectName         string
-	EnvironmentName     string
-	Version             string
-	PreviousVersion     string
-	isOpen              bool
-	ServiceFactory      ServiceFactory
-	ContainerFactory    ServiceFactory
-	DependenciesFactory DependenciesFactory
-	VolumesFactory      VolumesFactory
-	SecretsFactory      SecretsFactory
-	HostsFactory        HostsFactory
-	EnvironmentLookup   config.EnvironmentLookup
-	ResourceLookup      config.ResourceLookup
-	LoggerFactory       logger.Factory
-	Project             *Project
+	ComposeFiles         []string
+	ComposeBytes         [][]byte
+	ProjectName          string
+	EnvironmentName      string
+	Version              string
+	PreviousVersion      string
+	isOpen               bool
+	ServiceFactory       ServiceFactory
+	ContainerFactory     ServiceFactory
+	DependenciesFactory  DependenciesFactory
+	VolumesFactory       VolumesFactory
+	SecretsFactory       SecretsFactory
+	HostsFactory         HostsFactory
+	EnvironmentLookup    config.EnvironmentLookup
+	ResourceLookup       config.ResourceLookup
+	LoggerFactory        logger.Factory
+	Project              *Project
+	ExistingServiceNames map[string]string
```

Comment:
Nothing else on this context struct follows the pattern you are introducing here. It worries me, but I don't really know the code well enough to say where it should go yet.

Reply:
By "the pattern" I mean saving specific service state on this context.
Maybe the thing to do is to have a single function somewhere that gathers and deletes all the appropriate services